src/redux/categories/categoriesSlice.js

```diff
@@ -0,0 +1,15 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  categories: [],
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: () => 'under construction',
+  },
+});
+
+export default categoriesSlice.reducer;
```

Comment:
- [ ] Great job. But the checkStatus button must dispatch an action when it's clicked which is `Under construction` text. The above reducer might not be able to do that when you implement changes in the component. Take a look at below snippet.
- [ ] You might also need to export the action creator(method) -`checkStatus`

```
reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

Reply:
Hi @Qoosim, thank you for taking the time to review my project, I appreciate the feedback you provided to me. However, on the `checkStatus` method, when I'm trying to assign to the status categories property a value using the ternary operator as you advised:
``` sh
  state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
```
I'm getting the following linter error:
```sh
  12:7  error  Assignment to property of function parameter 'state'  no-param-reassign
```
To avoid it, I made the `checkStatus` method return either the `sate.categories` property or the `Under construction` text as follows:

```sh
    checkStatus: (state) => {
      if (state.categories.length) return state.categories;
      return 'Under construction';
    },
```